src/require.jl

```diff
@@ -1,35 +1,23 @@
-import Base: require
-
 export @require
 
 isprecompiling() = ccall(:jl_generating_output, Cint, ()) == 1
 
-# We are overwriting `Base.require` here, which is a pretty
-# horrible hack. Furthermore we need to not overwrite
-# `Base.require` while precompiling (thus @guard) and we need
-# to overwrite it in a world newer than the tasks in `oldcall.jl`
-# Since `Base.require` uses `eval` it switches automatically into
-# a newer world and call this `Base.require` from there.
-# We need to repeatedly switch back into the old world.
-@init @guard begin
-  function Base.require(mod::Symbol)
-    Main.Requires.oldcall(Base.require, mod)
-    Main.Requires.loadmod(string(mod))
-  end
+@init begin
+  push!(Base.package_callbacks, loadmod)
```

Comment:
Why? There are lots and lots of packages out there that have ``julia 0.6-`` in ``REQUIRE``, and I'm sure many of them rely on features that were introduced somewhere midway through the 0.6 cycle. How is this different here? Once julia 0.6 is released the version bound will be just ``0.6`` I assume in any case, right?